#VRML_SIM R2021a utf8
WorldInfo {
  info [
    "Arena for Stage 1 of SLRC2021 University Category"
  ]
  title "SLRC2021 UniS1"
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.43513332922711623 0.7834723251002135 0.44365538607999716 1.9834709322910091
  position 4.185188860768801 7.003432607624794 -0.6817156581118058
}
DirectionalLight {
  direction 0 -1 0
}
DirectionalLight {
  direction -1 -1 0
}
DirectionalLight {
  direction 0 -1 -1
}
DirectionalLight {
  direction 1 -1 0
}
DirectionalLight {
  direction 0 -1 1
}
RectangleArena {
  translation -0.02 0 0
  name "Floor_black"
  floorSize 3 4
  floorTileSize 2 2
  floorAppearance Parquetry {
    type "dark strip"
    colorOverride 0 0 0
  }
  wallAppearance DEF Wall_white PBRAppearance {
    roughness 1
    metalness 0
    name "Wall_white"
  }
}
DEF Line Solid {
  translation -0.613 -0.00485 -1.5
  scale 0.01 0.01 0.01
  children [
    Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Mesh {
        url [
          "../solidworks/Lines.STL"
        ]
      }
      castShadows FALSE
      isPickable FALSE
    }
  ]
  name "White_lines"
  locked TRUE
  translationStep 0
  rotationStep 0
}
DEF Squares Group {
  children [
    DEF Patch Solid {
      translation -0.075 -0.00484 1.1
      scale 0.01 0.01 0.01
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Mesh {
            url [
              "../solidworks/Square.STL"
            ]
          }
          castShadows FALSE
          isPickable FALSE
        }
      ]
      name "color_patch"
      locked TRUE
      translationStep 0
      rotationStep 0
    }
    DEF Green Solid {
      translation 0.01 -0.00484 -1.65
      scale 0.01 0.01 0.01
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Mesh {
            url [
              "../solidworks/Square.STL"
            ]
          }
          castShadows FALSE
          isPickable FALSE
        }
      ]
      name "green_square"
      locked TRUE
      translationStep 0
      rotationStep 0
    }
    DEF Red Solid {
      translation -0.41 -0.00484 -1.65
      scale 0.01 0.01 0.01
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Mesh {
            url [
              "../solidworks/Square.STL"
            ]
          }
          castShadows FALSE
          isPickable FALSE
        }
      ]
      name "red_square"
      locked TRUE
      translationStep 0
      rotationStep 0
    }
    DEF Blue Solid {
      translation 0.43 -0.00484 -1.65
      scale 0.01 0.01 0.01
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Mesh {
            url [
              "../solidworks/Square.STL"
            ]
          }
          castShadows FALSE
          isPickable FALSE
        }
      ]
      name "blue_square"
      locked TRUE
      translationStep 0
      rotationStep 0
    }
  ]
}
DEF Inner_wall Solid {
  translation 0.005 0 -0.66
  scale 0.01 0.01 0.01
  children [
    DEF Inner_wall Shape {
      appearance USE Wall_white
      geometry Mesh {
        url [
          "../solidworks/Inner_walls.STL"
        ]
      }
    }
  ]
  name "Inner_walls"
  boundingObject USE Inner_wall
  locked TRUE
  recognitionColors [
    1 1 1
  ]
  translationStep 0
  rotationStep 0
}
DEF Right_pillar Solid {
  translation 0.2 0.02 0.74
  children [
    Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.04 0.12
      }
    }
  ]
  name "R_pillar"
  radarCrossSection 1
}
DEF Left_pillar Solid {
  translation -0.2 0.04 0.74
  children [
    Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.09 0.12
      }
    }
  ]
  name "L_pillar"
  radarCrossSection 1
}
Robot {
  translation -4.35207e-06 0 1.64
  rotation 0 1 0 3.14159
  children [
    DEF body Transform {
      translation 0 0.02 0
      rotation 0 1 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.815686 1 0.388235
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.12 0.01 0.1
          }
        }
      ]
    }
    DEF right_wheel HingeJoint {
      jointParameters HingeJointParameters {
        position 1432.3199093411852
        anchor -0.055 0.02 -0.03
      }
      device [
        RotationalMotor {
          name "left_motor"
        }
      ]
      endPoint Solid {
        translation -0.05499999411369054 0.020003784941262315 -0.02999317517352416
        rotation -0.45704046766810147 -0.4570408402332355 0.7630384533381139 4.445181928186963
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 0.333333 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF left_wheel HingeJoint {
      jointParameters HingeJointParameters {
        position 1432.3199085819226
        anchor 0.055 0.02 -0.03
      }
      device [
        RotationalMotor {
          name "right_motor"
        }
      ]
      endPoint Solid {
        translation 0.055000005865753036 0.02000367381203341 -0.029993353572337626
        rotation -0.4570402385336936 -0.45704061109972277 0.7630387278288414 4.445182275153814
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0 0.333333 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF piller_detect HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0.045 0
      }
      device [
        RotationalMotor {
          name "piller_detect_bottom"
        }
      ]
      endPoint Solid {
        translation 0 0.045 0
        rotation 0 1 0 3.141592653589793
        children [
          DEF sensor Shape {
            appearance PBRAppearance {
              baseColor 1 0.666667 1
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.04
              radius 0.01
            }
          }
          DistanceSensor {
            translation 0 -0.015 0
            name "bottom_sensor"
            lookupTable [
              0 0 0
              0.4 1000 0
            ]
            type "infra-red"
          }
          DistanceSensor {
            name "top_sensor"
            lookupTable [
              0 0 0
              0.4 1000 0
            ]
            type "infra-red"
          }
        ]
        name "solid(3)"
        boundingObject USE sensor
        physics Physics {
        }
      }
    }
    DEF support_ball Solid {
      translation 0 0.0075 0.05
      rotation 0 0 -1 2.3276357876976306e-07
      children [
        DEF ball Shape {
          appearance PBRAppearance {
            roughness 1
            metalness 0
          }
          geometry Sphere {
            radius 0.0075
          }
        }
      ]
      name "solid(2)"
      boundingObject USE ball
      physics Physics {
      }
    }
    DEF left_2 DistanceSensor {
      translation 0.04 0.02 0.06
      rotation 0 0 1 -1.5707
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.666667 0 0.498039
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.008 0.008 0.008
          }
        }
      ]
      name "left2_ir"
      type "infra-red"
    }
    DEF left DistanceSensor {
      translation 0.02 0.02 0.06
      rotation 0 0 1 -1.5707
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0235294 0.0352941 0.85098
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.008 0.008 0.008
          }
        }
      ]
      name "left_ir"
      type "infra-red"
    }
    DEF center DistanceSensor {
      translation 0 0.02 0.06
      rotation 0 0 1 -1.5707
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0235294 0.0352941 0.85098
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.008 0.008 0.008
          }
        }
      ]
      name "center_ir"
      type "infra-red"
    }
    DEF right DistanceSensor {
      translation -0.02 0.02 0.06
      rotation 0 0 1 -1.5707
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0235294 0.0352941 0.85098
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.008 0.008 0.008
          }
        }
      ]
      name "right_ir"
      type "infra-red"
    }
    DEF right_2 DistanceSensor {
      translation -0.04 0.02 0.06
      rotation 0 0 1 -1.5707
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.666667 0 0.498039
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.008 0.008 0.008
          }
        }
      ]
      name "right2_ir"
      type "infra-red"
    }
    Camera {
      translation 0 0.02 0.06
      rotation 1 0 0 -1.5707996938995747
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.01 0.01 0.01
              }
            }
          ]
        }
      ]
      recognitionColors [
        0 0 0
      ]
    }
  ]
  boundingObject USE body
  physics Physics {
    mass 1
  }
  controller "my_controller"
  controllerArgs [
    ""
  ]
}
